### DISCUSSION TITLE
Cannot generate conformers for a spiral molecule

### DISCUSSION THREAD

**Discussion by iuhgnor** [rdkit/rdkit#5841] [Q&A] [answered]
RDKit fails to generate conformers for the spiral molecule and it takes a long time (more than 1 min). How to get the conformers? Thanks.

```python
print(rdkit.__version__)
smi = 'C1=CC=C2C(=C1)C=CC3=C2C4=C(C=CC5=C4C6=C(C=C5)C=CC7=C6C8=C(C=CC9=C8C1=C(C=C9)C=CC2=CC=CC=C21)C=C7)C=C3'
mol = Chem.MolFromSmiles(smi)
mol = Chem.AddHs(mol)
AllChem.EmbedMolecule(mol)
mol.GetNumConformers()
```

Output:
```python
2022.03.5
0
```

**Comment by greglandrum** [comment, marked as answer]
@iuhgnor The usual solution when the standard conformer generation doesn't work is to try and use random coordinate embedding instead of the standard metric matrix embedding. Unfortunately that doesn't work here:
```
In [7]: from rdkit.Chem import rdDistGeom

In [8]: mol = Chem.AddHs(mol)

In [9]: ps = rdDistGeom.ETKDGv3()

In [10]: ps.useRandomCoords = True

In [11]: rdDistGeom.EmbedMolecule(mol,ps)
Out[11]: -1
```
-1 indicates that the conformer generation failed.

This particular molecule is failing because the "basic knowledge" (K) terms in the RDKit's ETKDG conformer generator, which try to force sp2 centers to be planar, cannot be adequately satisified. This make sense for this molecule.

Since you can't currently control the tolerence threshold for the K terms, your best bet is to just turn them off for this molecule:
```
In [13]: ps = rdDistGeom.ETKDGv3()

In [14]: ps.useBasicKnowledge = False

In [15]: ps.useRandomCoords = True

In [16]: rdDistGeom.EmbedMolecule(mol,ps)
Out[16]: 0
```
Since this conformer was generated without the K terms, it's going to be ugly, so you probably want to clean it up with a force-field minimization:
```
In [19]: from rdkit.Chem import rdForceFieldHelpers

In [20]: rdForceFieldHelpers.MMFFOptimizeMolecule(mol)
Out[20]: 0
```

**Reply by iuhgnor** [reply]
Thanks for your detailed reply.
